src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -108,6 +108,14 @@ public Task StartReceive(CancellationToken cancellationToken = default)
             return Task.CompletedTask;
         }
 
+        public async Task ChangeConcurrency(PushRuntimeSettings limitations,
+            CancellationToken cancellationToken = new CancellationToken())
+        {
+            await StopReceive(cancellationToken).ConfigureAwait(false);
+            await Initialize(limitations, onMessage, onError, cancellationToken).ConfigureAwait(false);
```

Comment:
This will cause queue purging to happen if `PurgeOnStartup` is true

Reply:
True, fixed!